tests/components/Microsoft.DotNet.UpgradeAssistant.MSBuild.Tests/TargetFrameworkIdentifierTests.cs

```diff
@@ -80,5 +82,32 @@ public void TfmCompare(string target, string tfm, int expected)
             static TargetFrameworkMoniker? Create(string? input)
                 => input is null ? null : new TargetFrameworkMoniker(input);
         }
+
+        [InlineData("net6.0-windows", "net5.0-windows", "net6.0")]
+        [InlineData("net6.0-windows", "net6.0", "net5.0-windows")]
+        [InlineData("net6.0-windows", "net5.0", "net6.0-windows")]
+        [InlineData("net6.0-windows10.0.5", "net5.0", "net6.0-windows10.0.5")]
+        [InlineData("net6.0-windows10.0.5", "net5.0-windows10.0.5", "net6.0")]
+        [InlineData("net6.0-windows10.1.5", "net5.0-windows10.1.5", "net6.0-windows10.0.5")]
```

Comment:
hmm, Why would the expected be net6.0-windows10.1.5? Should it not be net6.0-windows10.0.5, coz that is the highest version (combination of name and version) there might not be a 10.1.5 for 6.0, is it not?

Reply:
When merging, you want to take the highest framework version (here that's net6.0) and the highest platform version (windows10.1.5)